src/status_im2/contexts/onboarding/common/navigation_bar/view.cljs

```diff
@@ -14,6 +14,12 @@
      :left-section          {:type                :blur-bg
                              :icon                :i/arrow-left
                              :icon-override-theme :dark
-                             :on-press            (when-not disable-back-button?
-                                                    #(rf/dispatch [:navigate-back]))}
+                             :on-press            (fn []
+                                                    (if override-left-button-on-press
+                                                      (override-left-button-on-press)
+                                                      (when-not disable-back-button?
+                                                        (rf/dispatch (if stack-id
+                                                                       [:navigate-back-within-stack
+                                                                        stack-id]
+                                                                       [:navigate-back])))))}
```

Comment:
a bit hard to read because of how it's formatted — could we do something about it? e.g. moving it all to a separate function

Reply:
Done